Improve styling of job history table

DuMaM opened the pull request:
This plugin base on many different sub pages. 
I want to update style for job config history. 
This will set up direction where those tables should be converted.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

<!--
Put an `x` into the [ ] to show you have filled the information.
The template comes from https://github.com/jenkinsci/.github/blob/master/.github/pull_request_template.md 
You can override it by creating .github/pull_request_template.md  in your own repository 
-->

![image](https://user-images.githubusercontent.com/9051964/172248651-9f25fc76-42f1-4c93-9689-e48c96260db7.png)

DuMaM commented:
Current problems:
* Somehow page switching stopped working :( 
* I don't know how to add submission for newer button style. 
* Every second "restore" button does not showing up

DuMaM commented:
I found problem connected with page switch. It detects submission when there is no url attached to it and then calculate diff File.

DuMaM commented:
Ok I fixed button problems. Now they do not make any submission, so everything works as expected.
Whats left is a "restore button" and positioning.

DuMaM commented:
![image](https://user-images.githubusercontent.com/9051964/172069329-d528cfd5-17f6-4671-86e6-53ea898c85bb.png)
Current progress:
* buttons positioning got fixed 
* Show Diff submission stopped working :/
* Button themselves should act as a hrefs
* Still I didn't found a reason why "restore" icon do not show it self

DuMaM commented:
Current progress:
![image](https://user-images.githubusercontent.com/9051964/172227961-ad520ee9-c747-459d-944c-89af8eeb33b1.png)

Current problems:
* `all` button do not highlight
* `restore` icon do not show itself for all config changes

DuMaM commented:
Only restore button problem left.

DuMaM commented:
![image](https://user-images.githubusercontent.com/9051964/172248651-9f25fc76-42f1-4c93-9689-e48c96260db7.png)
Everything now works 😉

NotMyFault commented:
Looks very nice, thanks for your contribution!

I'm wondering what's up with the radio buttons, they seem to be not aligned in the table center and move around if you click on them:
https://user-images.githubusercontent.com/13383509/172365618-2522a607-47fd-40bb-af86-1088d7070a7d.mp4

NotMyFault commented:
Looks very nice, thanks for your contribution!

I'm wondering what's up with the radio buttons, they seem to be not aligned in the table center and move around if you click on them:
https://user-images.githubusercontent.com/13383509/172365618-2522a607-47fd-40bb-af86-1088d7070a7d.mp4

NotMyFault commented:
> > I'm wondering what's up with the radio buttons, they seem to be not aligned in the table center and move around if you click on them: https://user-images.githubusercontent.com/13383509/172365618-2522a607-47fd-40bb-af86-1088d7070a7d.mp4
> 
> Thats the case for all radios in this jenkins. I got similar experience in other plugins.

Oic, that should probably be reported on Jira as core issue, if it affects several plugins.

NotMyFault commented on src/main/resources/hudson/plugins/jobConfigHistory/JobConfigHistoryProjectAction/index.jelly:
```diff
@@ -123,24 +115,27 @@
+                          <td style="text-align:center">
+                            <j:if test="${it.hasDeleteEntryPermission()}">
+                              <j:set var="message" value="${%Do you really want to delete the history entry} "/>
+                              <button type="button" class="jenkins-button jenkins-button--transparent" onClick="removeEntryFromTable('table-row-${configNr}', '${config.date}', 'null', '${message}')" value="X">
```
```suggestion
                              <button type="button" class="jenkins-button jenkins-button--destructive" onClick="removeEntryFromTable('table-row-${configNr}', '${config.date}', 'null', '${message}')" value="X">

```
A red button may hightlight a destructive action better.

DuMaM replied:
Good idea :)

DuMaM commented:
@NotMyFault Is it ok to use transparent button for '<' '>' when they are inactive?
Maybe normal one will be better?

NotMyFault commented:
> @NotMyFault Is it ok to use transparent button for '<' '>' when they are inactive?

Where do you mean exactly?

DuMaM commented:
> > @NotMyFault Is it ok to use transparent button for '<' '>' when they are inactive?
> 
> Where do you mean exactly?

Inactive (no page 0)
![image](https://user-images.githubusercontent.com/9051964/173043803-b445bdab-c1f8-4048-bdb5-8888d013d2c9.png)

Active (there is previous page -> 1)
![image](https://user-images.githubusercontent.com/9051964/173043948-3135d30c-362b-4fa3-bcae-0061f66ad1cd.png)

NotMyFault commented:
Oic, yeah I agree, the normal button looks better.

DuMaM commented:
Is there a way to show inactive button? If i use a normal one then people may get confused

NotMyFault commented:
> Is there a way to show inactive button? If i use a normal one then people may get confused

The normal one as seen on
![](https://user-images.githubusercontent.com/9051964/173043948-3135d30c-362b-4fa3-bcae-0061f66ad1cd.png)
looks like any other modern button in Jenkins.

I'm unsure if choosing a different style might break with this scheme 🤔

DuMaM commented:
![image](https://user-images.githubusercontent.com/9051964/173203559-f532d2ad-052c-48e5-bf33-2d0531351171.png)
I'm pretty happy with current result :)

DuMaM commented:
@NotMyFault I fixed urls for Gitpod now. At least for pages I was working with.
Also I updated one more site (root one).
![image](https://user-images.githubusercontent.com/9051964/173206793-f13fc8d5-c8d5-489f-bb0f-b0d858e47d77.png)
There was duplicated filter functionality implemented as urls in `Main-panel` buttons and `Side-panels` tasks, so i removed those main ones.

NotMyFault commented:
> @NotMyFault I fixed urls for Gitpod now. At least for pages I was working with.

Yup, all looking fine so far 👏🏻 

Though the table headers aren't set and follow your scroll behavior:
https://user-images.githubusercontent.com/13383509/173225019-67294d79-1d91-4d1e-a9c9-00de0461ed83.mp4

DuMaM commented:
> > @NotMyFault I fixed urls for Gitpod now. At least for pages I was working with.
> 
> Yup, all looking fine so far 👏🏻
> 
> Though the table headers aren't set and follow your scroll behavior: https://user-images.githubusercontent.com/13383509/173225019-67294d79-1d91-4d1e-a9c9-00de0461ed83.mp4

Headers fixed :)

NotMyFault reviewed (approved):
Thanks a bunch for your contribution, nice to see the modernized jenkins tables here as well (at least at this specific history view).
